### ISSUE TITLE
Re-structure package/publish structure as monorepo

### ISSUE THREAD

**Issue by gandlafbtc** [cashubtc/cashu-ts#231]
For improved DX, we should migrate to using a monorepo and consolidate the cashu-crypto-ts repo.

we could use lerna: https://lerna.js.org/

**Comment by Egge21M** [comment]
Yes, let's use this thread to discuss possible structures. The amount of assets that we are building is quite low, so I personally do not think that we need to introduce any new dependencies to the project. Here is what I had in mind:

- include the crypto-package in the main repo
- replace the npm-hosted dependency with a link of the local one
- use npm workspaces to bundle the crypto-package before bundling the library

That way we can still publish the crypto-package as a separate module if desirable, but make sure that cashu-ts always uses the latests version by default and developers can do all changes in a single repo.

We could also use git submodules to keep git history separated, but I dont think that is a good thing in our case and slows down development. Instead we could come up with a rule that all commits for the crypto lib should be squashed and prepended with `crypto:`
